lib/cc/analyzer/engines_config_builder.rb

```diff
@@ -56,20 +58,16 @@ def names_and_raw_engine_configs
         end
       end
 
-      def include_paths
-        IncludePathsBuilder.new(exclude_paths, Array(@requested_paths)).build
-      end
-
-      def exclude_paths
-        PathPatterns.new(@config.exclude_paths || []).expanded +
-          gitignore_paths
+      def workspace
+        @workspace ||= Workspace.new(paths: Array(@requested_paths)).filter(@config.exclude_paths)
       end
 
-      def gitignore_paths
-        if File.exist?(".gitignore")
-          `git ls-files --others -i -z --exclude-from .gitignore`.split("\0")
-        else
-          []
+      # The yaml gem turns a config file string into a hash, but engines expect
+      # the string. So we (for now) need to turn it into a string in that one
+      # scenario.
+      def normalize_config_file(config)
```

Comment:
Actually, thinking on it, it feels like this fix belongs in cc-yaml: if we intend to allow `config: file-name`, then `cc-yaml` should not presume the type of the `config` key.